atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/expectations/TrackingKeyValueServiceTest.java

```diff
@@ -0,0 +1,435 @@
+    private static final int CELL_BATCH_HINT = 2;
+    private static final byte[] START_BYTES = TrackingKeyValueServiceTestUtils.createBytesWithSize(5, (byte) 1);
+    private static final byte[] END_BYTES = TrackingKeyValueServiceTestUtils.createBytesWithSize(10, (byte) 1);
+    private static final ColumnSelection COLUMN_SELECTION = ColumnSelection.create(ImmutableSet.of(
+            TrackingKeyValueServiceTestUtils.createBytesWithSize(100, (byte) 0),
+            TrackingKeyValueServiceTestUtils.createBytesWithSize(101, (byte) 1),
+            TrackingKeyValueServiceTestUtils.createBytesWithSize(102, (byte) 2)));
+    private static final ColumnRangeSelection COLUMN_RANGE_SELECTION = new ColumnRangeSelection(START_BYTES, END_BYTES);
+    private static final BatchColumnRangeSelection BATCH_COLUMN_RANGE_SELECTION =
+            BatchColumnRangeSelection.create(COLUMN_RANGE_SELECTION, CELL_BATCH_HINT);
+    private static final CandidateCellForSweepingRequest CANDIDATE_CELL_FOR_SWEEPING_REQUEST =
+            ImmutableCandidateCellForSweepingRequest.builder()
+                    .startRowInclusive()
+                    .maxTimestampExclusive(TIMESTAMP)
+                    .shouldCheckIfLatestValueIsEmpty(false)
+                    .shouldDeleteGarbageCollectionSentinels(false)
+                    .build();
+    private static final Set<Cell> CELLS = ImmutableSet.of(
+            TrackingKeyValueServiceTestUtils.createCellWithSize(10, (byte) 0),
+            TrackingKeyValueServiceTestUtils.createCellWithSize(20, (byte) 1),
+            TrackingKeyValueServiceTestUtils.createCellWithSize(30, (byte) 2));
+
+    @Parameterized.Parameters(name = "size = {0}")
+    public static Object[] size() {
+        // divisible by 12 to allow creation of TableReference of wanted sizes
+        return new Object[] {12 * 129, 12 * 365, 12 * 411};
+    }
+
+    @Rule
+    public MockitoRule rule = MockitoJUnit.rule();
+
+    private final int size;
+    private final byte[] bytesOfSize;
+    private final ImmutableSet<TableReference> tableRefsOfSize;
+    private final ImmutableMap<Cell, Long> timestampByCellMapOfSize;
+    private final ImmutableList<byte[]> byteArrayListOfSize;
+    private final ImmutableMap<Cell, Value> backingValueByCellMapOfSize;
+    private final ImmutableMap<byte[], ImmutableMap<Cell, Value>> backingRowColumnRangeMapOfSize;
+    private final ImmutableList<RowResult<Value>> backingValueRowResultListOfSize;
+    private final ImmutableList<RowResult<Set<Long>>> backingLongSetRowResultListOfSize;
+    private final ImmutableList<ImmutableList<CandidateCellForSweeping>> backingCandidateForSweepingTableOfSize;
+    private final ImmutableMap<RangeRequest, TokenBackedBasicResultsPage<RowResult<Value>, byte[]>>
+            pageByRangeRequestMapOfSize;
+    private final ImmutableMultimap<Cell, Long> longByCellMultimapOfSize;
+    private final ImmutableMap<TableReference, byte[]> metadataForTablesOfSize;
```

Comment:
Note for readability: local variables ending with `OfSize` have byte size equal to `this.size`

Whereas the static objects have no size constraints